MayssaraOmar/assignment3/Word search/src/Word search.cpp

```diff
@@ -0,0 +1,48 @@
+#include <map>
+#include "word search.h"
+using namespace std;
+
+void get_words_from_grid(vector<vector<char> > &grid, int row, int col,
+		string word, map<pair<int, int>, bool> &vis,
+		map<pair<pair<int, int>, string>, bool> &memo, dictionary &dic,
```

Comment:
"memo" is not very clear in this context: what is this supposed to represent?
You can add this as a comment to the function.

Reply:
For example the following grid with the word "match" in dictionary:
a a r
t m d
c h x
We can obtain the word "match" by 2 ways:
1- by going from 'm' to 'a' upwards and then 't', 'c', 'h'
2- by going from 'm' to 'a' diagonally and then 't', 'c', 'h'
Here the ('t', 'c', 'h') is the repeated work as we call the function with the same input (row = 1, col = 0, word = "ca") twice.
We avoid repeating work by marking (1,0,"ca") true, in the second call, (1,0,"ca") will be true and the function will return.